File: BackEnd_API/API_BookStoreApp/BusinessObject/Models/Book.cs
using System;
using System.Collections.Generic;

namespace BusinessObject.Models;

public partial class Book
{
    public int BookId { get; set; }

    public string Title { get; set; } = null!;

    public int AuthorId { get; set; }

    public int CategoryId { get; set; }

    public string? Description { get; set; }

    public decimal Price { get; set; }

    public int? Stock { get; set; }

    public string? ImageUrl { get; set; }

    public DateTime? CreatedAt { get; set; }

    public virtual Author Author { get; set; } = null!;

    public virtual Category Category { get; set; } = null!;

    public virtual ICollection<OrderItem> OrderItems { get; set; } = new List<OrderItem>();
}

File: StudyTrackerApp/app/src/main/java/com/example/studytrackerapp/Models/Book.java
package com.example.studytrackerapp.Models;

public class Book {
    public int bookID;
    public String title;
    public int authorId;
    public int categoryId;
    public String description;
    public double price;
    public Integer stock; // nullable nên dùng Integer thay vì int
    public String imageUrl;
}

